### PR TITLE
Feature/esp lcd touch support multiple interfaces

### PR DESCRIPTION
**Pull request by espzav**

### CODE REVIEW

File: components/esp_lcd_touch/idf_component.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-version: "0.9.1"
+version: "0.9.2"
```
Comment (tore-espressif) [inline components/esp_lcd_touch/idf_component.yml:1]: There are breaking changes introduced; let's bump versions to 1.0.0

File: components/esp_lcd_touch/idf_component.yml [hunk 2]
```diff
@@ -1,4 +1,4 @@
...
 description: ESP LCD Touch - main component for using touch screen controllers
 url: https://github.com/espressif/esp-bsp/tree/master/components/esp_lcd_touch
 dependencies:
```
Comment (tore-espressif) [inline components/esp_lcd_touch/idf_component.yml:4]: IDF version >= 4.4.2 ?

File: components/esp_lcd_touch_ft5x06/esp_lcd_touch_ft5x06.c [hunk 3]
```diff
@@ -290,32 +282,19 @@ static esp_err_t touch_ft5x06_init(esp_lcd_touch_handle_t tp)
...
 
-    ft5x06_dev = &tp->config.device.i2c;
-    assert(ft5x06_dev->port >= 0 && ft5x06_dev->port < SOC_I2C_NUM);
-
-    /* Write register and data number */
-    return i2c_master_write_to_device(ft5x06_dev->port, ESP_LCD_TOUCH_FT5x06_I2C_ADDRESS, buff, sizeof(buff), ESP_LCD_TOUCH_FT5x06_I2C_TIMEOUT_MS / portTICK_PERIOD_MS);
+    // *INDENT-OFF*
+    /* Write data */
+    return esp_lcd_panel_io_tx_param(tp->io, reg, (uint8_t[]){data}, 1);
+    // *INDENT-ON*
```
Comment (tore-espressif) [inline components/esp_lcd_touch_ft5x06/esp_lcd_touch_ft5x06.c:290]: Nitpick: single line formating disable?

```suggestion
    /* Write data */
    return esp_lcd_panel_io_tx_param(tp->io, reg, (uint8_t[]){data}, 1);  // *INDENT-OFF*
```

File: components/esp_lcd_touch_gt911/esp_lcd_touch_gt911.c [hunk 4]
```diff
@@ -250,31 +246,19 @@ static void touch_gt911_read_cfg(esp_lcd_touch_handle_t tp)
...
     assert(tp != NULL);
 
-    gt911_dev = &tp->config.device.i2c;
-    assert(gt911_dev->port >= 0 && gt911_dev->port < SOC_I2C_NUM);
-
-    return i2c_master_write_to_device(gt911_dev->port, ESP_LCD_TOUCH_GT911_I2C_ADDRESS, reg_buf, sizeof(reg_buf), ESP_LCD_TOUCH_GT911_I2C_TIMEOUT_MS / portTICK_PERIOD_MS);
+    // *INDENT-OFF*
+    /* Write data */
+    return esp_lcd_panel_io_tx_param(tp->io, reg, (uint8_t[]){data}, 1);
+    // *INDENT-ON*
```
Comment (tore-espressif) [inline components/esp_lcd_touch_gt911/esp_lcd_touch_gt911.c:263]: Likewise: could single-line formatting disable used?

**Review by tore-espressif — CHANGES_REQUESTED** [review]
Nice work @espzav . I guess we need to wait until public esp-idf `master` and `release/v4.4` branches are updated, so we could pass the CI

File: components/esp_lcd_touch/idf_component.yml (hunk 2 above)
Comment (espzav) [reply components/esp_lcd_touch/idf_component.yml:4]: Right, done.

File: components/esp_lcd_touch_ft5x06/esp_lcd_touch_ft5x06.c (hunk 3 above)
Comment (espzav) [reply components/esp_lcd_touch_ft5x06/esp_lcd_touch_ft5x06.c:290]: This is not working...

File: components/esp_lcd_touch_gt911/esp_lcd_touch_gt911.c (hunk 4 above)
Comment (espzav) [reply components/esp_lcd_touch_gt911/esp_lcd_touch_gt911.c:263]: This is not working...

File: components/lcd_touch/esp_lcd_touch_ft5x06/include/esp_lcd_touch_ft5x06.h [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+esp_err_t esp_lcd_touch_new_i2c_ft5x06(const esp_lcd_panel_io_handle_t io, const esp_lcd_touch_config_t *config, esp_lcd_touch_handle_t *out_touch);
+
+/**
+ * @brief I2C address of the FT5x06 controller
+ *
+ */
+#define ESP_LCD_TOUCH_IO_I2C_FT5x06_ADDRESS (0x38)
+
+/**
+ * @brief Communication I2C device IO stricture
```
Comment (tore-espressif) [inline components/lcd_touch/esp_lcd_touch_ft5x06/include/esp_lcd_touch_ft5x06.h:36]:
```suggestion
 * @brief Touch IO configuration structure
```

(likewise in other touch header files)

**Review by tore-espressif — APPROVED** [review]
@espzav I left few nitpicks. Look good overall!

File: components/lcd_touch/esp_lcd_touch_ft5x06/README.md [hunk 6]
```diff
@@ -21,6 +21,8 @@ Alternatively, you can create `idf_component.yml`. More is in [Espressif's docum
 Initialization of the touch component.
 
 ```
```
[the remark this answers is not in the document]
Comment (espzav) [reply components/lcd_touch/esp_lcd_touch_ft5x06/README.md:23]: It is not tested and we cannot test now. But we can add something like "It should work on SPI too" or somethink like that.

**Pull request merged**